### PR TITLE
Add presentableVarsState for debug output with caching

### PR DESCRIPTION
**Pull request by nikolay-egorov**

### CODE REVIEW

**Comment by ileasile** [comment]
Not sure if we really need one more entity for almost the same thing...

**Comment by nikolay-egorov** [comment]
I suggest to leave plain realisation of map recreation in Notebook, without caching. Like 
```
        val ans = mutableMapOf<String, Result<Any?>>()
        ans.putAll(
            varsState.mapValues {
                it.value.value
            }
        )
        return ans
```
 It's way convinient to keet it here than to assamble this call in bytecode

**Comment by nikolay-egorov** [comment]
I changed api, introduced private object for debub purposes in repl

File: src/main/kotlin/org/jetbrains/kotlinx/jupyter/repl.kt [hunk 1]
```diff
@@ -357,6 +357,8 @@ class ReplForJupyterImpl(
         internalVariablesMarkersProcessor,
     )
 
+    private val debugUtilityProvider = DebugUtils
```
Comment (ileasile) [inline src/main/kotlin/org/jetbrains/kotlinx/jupyter/repl.kt:360]: Maybe make the property name consistent with object/class name?

File: src/main/kotlin/org/jetbrains/kotlinx/jupyter/debugUtils.kt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package org.jetbrains.kotlinx.jupyter
+
+import org.jetbrains.kotlinx.jupyter.api.Notebook
+
+object DebugUtils {
```
[the remark this answers is not in the document]
Comment (nikolay-egorov) [reply src/main/kotlin/org/jetbrains/kotlinx/jupyter/debugUtils.kt:5]: Sounds good, for now I've added this way, but it would be clear as time goes by

**Pull request merged**